addons/sale_project/models/product.py

```diff
@@ -5,17 +5,21 @@
 from odoo.exceptions import ValidationError
 
 
-SERVICE_POLICY = [
-    # (service_policy, string)
-    ('ordered_prepaid', 'Prepaid/Fixed Price'),
-    ('delivered_milestones', 'Based on Milestones'),
-    ('delivered_manual', 'Based on Delivered Quantity (Manual)'),
-]
-
-
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
+    @api.model
+    def _selection_service_policy(self):
+        service_policies = [
+            # (service_policy, string)
+            ('ordered_prepaid', 'Prepaid/Fixed Price'),
+            ('delivered_milestones', 'Based on Milestones'),
+            ('delivered_manual', 'Based on Delivered Quantity (Manual)'),
+        ]
+        if not self.user_has_groups('project.group_project_milestone'):
+            service_policies = [policy for policy in service_policies if policy[0] != 'delivered_milestones']
```

Comment:
Wouldn't it be cleaner like this?
```suggestion
        service_policies = [
            # (service_policy, string)
            ('ordered_prepaid', 'Prepaid/Fixed Price'),
            ('delivered_manual', 'Based on Delivered Quantity (Manual)'),
        ]
        if self.user_has_groups('project.group_project_milestone'):
            service_policies.insert(1, ('delivered_milestones', 'Based on Milestones'))
```